be/src/exec/vectorized/aggregator.cpp

```diff
@@ -299,7 +300,20 @@ Status Aggregator::prepare(RuntimeState* state, ObjectPool* pool, RuntimeProfile
     return Status::OK();
 }
 
-Status Aggregator::reset_state(std::vector<vectorized::ChunkPtr>&& chunks) {
+Status Aggregator::reset_state(starrocks::RuntimeState* state, const std::vector<vectorized::ChunkPtr>& chunks,
+                               pipeline::Operator* op) {
+    RETURN_IF_ERROR(_reset_state(state));
+    begin_pending_reset_state();
```

Comment:
during begin and end pending state, do we need the lock?

Reply:
no, it just tells aggregator, the refill-back chunks are intermediate type, so aggregator should invoke merge method of agg func to process these chunks. there is no concurrent issue.